### PR TITLE
[Keyboard] Add YMDK NP24 (ATmega32U4 with 6 RGB LEDs)

### PR DESCRIPTION
**Pull request by nomis**
24 key number-pad with 6 RGB LEDs on the bottom (black PCB) sold by YMDK

## Description

Not to be confused with the 8 RGB LED version (white PCB) that has an ATmega32A and is already handled by "ymdk_np21".

Given how many potential variants there are for a 6x4 keypad I've put the microcontroller and LED count in the name.

The original firmware USB device looks like this:
  New USB device found, idVendor=feed, idProduct=6060, bcdDevice= 0.01
  New USB device strings: Mfr=1, Product=2, SerialNumber=0
  Product: keyboard
  Manufacturer: qmkbuilder

It has the standard Atmel DFU bootloader:
  New USB device found, idVendor=03eb, idProduct=2ff4, bcdDevice= 0.00
  New USB device strings: Mfr=1, Product=2, SerialNumber=3
  Product: ATm32U4DFU
  Manufacturer: ATMEL
  SerialNumber: 1.0.0

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by nomis** [comment]
The original layout supplied is a numpad with the extra top row:
F1, F2, F3, Fn

and the function layer for the top row is:
RGB_MOD, BL_STEP, __, __

There's nothing else on the function layer for any of the other keys.

File: keyboards/ymdk/np24/u4rgb6/rules.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/ymdk/np24/u4rgb6/rules.mk:11]:
```suggestion
```

File: keyboards/ymdk/np24/u4rgb6/u4rgb6.c [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "u4rgb6.h"
```
Comment (drashna) [inline keyboards/ymdk/np24/u4rgb6/u4rgb6.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ymdk/np24/u4rgb6/u4rgb6.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+) { \
+    { k00, k10, k20, k30 }, \
+    { k01, k11, k21, k31 }, \
+    { k02, k12, k22, XXX }, \
+    { k03, k13, k23, k33 }, \
+    { k04, k14, k24, XXX }, \
+    { XXX, k15, k25, k35 }  \
+}
+
+#define LAYOUT LAYOUT_ortho_6x4
```
Comment (drashna) [inline keyboards/ymdk/np24/u4rgb6/u4rgb6.h:69]:
```suggestion
```

File: keyboards/ymdk/np24/u4rgb6/rules.mk [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no    # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no     # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
```
Comment (drashna) [inline keyboards/ymdk/np24/u4rgb6/rules.mk:24]:
```suggestion
RGBLIGHT_ENABLE = yes

LAYOUTS = ortho_6x4 numpad_6x4 ortho_4x6
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by nomis** [comment]
(I've made the changes requested)

At some point `ymdk_np21` could be moved to something like `ymdk/np24/a_rgb8/`.

There's also `ymd75` and `ymd96` at the top level.

File: keyboards/ymdk/np24/u4rgb6/rules.mk [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
```
Comment (fauxpark) [inline keyboards/ymdk/np24/u4rgb6/rules.mk:17]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ymdk/np24/u4rgb6/readme.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+Not to be confused with the 8 RGB LED version (white PCB) that has an ATmega32A ("ymdk_np21").
+
+* Keyboard Maintainer: [QMK Community](https://github.com/qmk)
+* Hardware Supported: ATmega32U4
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ymdk/np24rgb:default
+
+Flashing example for this keyboard ([after setting up the bootloadHID flashing environment](https://docs.qmk.fm/#/flashing_bootloadhid))
```
Comment (fauxpark) [inline keyboards/ymdk/np24/u4rgb6/readme.md:14]:
```suggestion
Flashing example for this keyboard:
```

File: keyboards/ymdk/np24/u4rgb6/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+    make ymdk/np24rgb:default:flash
+
+**Reset Key**: Hold down the 'Top Left Key' (USB on top) while plugging in the keyboard, or press the button on the bottom of the PCB.
+
```
Comment (fauxpark) [inline keyboards/ymdk/np24/u4rgb6/readme.md:19]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**